tests/Stripe/StripeObjectTest.php

```diff
@@ -147,38 +176,234 @@ public function testSerializeParametersOnArrayThatLengthens()
+        $deepCopyReflector->setAccessible(true);
+
+        // This is used to access the `_opts` protected variable
+        $optsReflector = new \ReflectionProperty('Stripe\\StripeObject', '_opts');
+        $optsReflector->setAccessible(true);
+
+        $opts = [
+            "api_base" => Stripe::$apiBase,
+            "api_key" => "apikey",
+        ];
+        $values = [
+            "id" => 1,
+            "name" => "Stripe",
+            "arr" => [
+                StripeObject::constructFrom(["id" => "index0"], $opts),
+                "index1",
+                2,
+            ],
+            "map" => [
+                "0" => StripeObject::constructFrom(["id" => "index0"], $opts),
+                "1" => "index1",
+                "2" => 2
+            ],
+        ];
+        $copyValues = $deepCopyReflector->invoke(null, $values);
+        // we can't compare the hashes directly because they have embedded
+        // objects which are different from each other
+        $this->assertEquals($values["id"], $copyValues["id"]);
+        $this->assertEquals($values["name"], $copyValues["name"]);
+        $this->assertEquals(count($values["arr"]), count($copyValues["arr"]));
+        // internal values of the copied StripeObject should be the same,
+        // but the object itself should be new (hence the assertNotSame)
+        $this->assertEquals($values["arr"][0]["id"], $copyValues["arr"][0]["id"]);
+        $this->assertNotSame($values["arr"][0], $copyValues["arr"][0]);
+        // likewise, the Util\RequestOptions instance in _opts should have
+        // copied values but be a new instance
+        $this->assertEquals(
+            $optsReflector->getValue($values["arr"][0]),
+            $optsReflector->getValue($copyValues["arr"][0])
+        );
+        $this->assertNotSame(
+            $optsReflector->getValue($values["arr"][0]),
+            $optsReflector->getValue($copyValues["arr"][0])
+        );
+        // scalars however, can be compared
+        $this->assertEquals($values["arr"][1], $copyValues["arr"][1]);
+        $this->assertEquals($values["arr"][2], $copyValues["arr"][2]);
+        // and a similar story with the hash
+        $this->assertEquals($values["map"]["0"]["id"], $copyValues["map"]["0"]["id"]);
+        $this->assertNotSame($values["map"]["0"], $copyValues["map"]["0"]);
+        $this->assertNotSame(
+            $optsReflector->getValue($values["arr"][0]),
+            $optsReflector->getValue($copyValues["arr"][0])
+        );
+        $this->assertEquals(
+            $optsReflector->getValue($values["map"]["0"]),
+            $optsReflector->getValue($copyValues["map"]["0"])
+        );
+        $this->assertNotSame(
+            $optsReflector->getValue($values["map"]["0"]),
+            $optsReflector->getValue($copyValues["map"]["0"])
+        );
+        $this->assertEquals($values["map"]["1"], $copyValues["map"]["1"]);
+        $this->assertEquals($values["map"]["2"], $copyValues["map"]["2"]);
```

Comment:
Nit, but the whole section above is pretty dense. I like that we have good test coverage, but it could probably use some more whitespace for readability.